pennylane/data/base/attribute.py

```diff
@@ -398,21 +394,21 @@ def __init_subclass__(  # pylint: disable=arguments-differ
             existing_type = DatasetAttribute.type_consumer_registry.get(type_)
             if existing_type is not None:
                 warnings.warn(
-                    f"Conflicting default types: Both '{cls}' and '{existing_type}' consume type '{type_}'. '{type_}' will now be consumed by '{cls}'"
+                    f"Conflicting default types: Both '{cls.__name__}' and '{existing_type.__name__}'"
+                    f" consume type '{type_.__name__}'. '{type_.__name__}'"
+                    f" will now be consumed by '{cls.__name__}'"
```

Comment:
PennyLane style preference here:
```suggestion
                    f"Conflicting default types: Both '{cls.__name__}' and '{existing_type.__name__}' "
                    f"consume type '{type_.__name__}'. '{type_.__name__}' "
                    f"will now be consumed by '{cls.__name__}'"
```